scripts/update_changelog_and_credits.py

```diff
@@ -0,0 +1,452 @@
+            return
+
+
+def main():
+    """Collects necessary info and dumps it to disk."""
+    branch_name = common.get_current_branch_name()
+    if not re.match(r'release-\d+\.\d+\.\d+$', branch_name):
+        raise Exception(
+            'This script should only be run from the latest release branch.')
+
+    if not os.path.exists(feconf.RELEASE_SUMMARY_FILEPATH):
+        raise Exception(
+            'Release summary file %s is missing. Please run the '
+            'release_info.py script and re-run this script.' % (
+                feconf.RELEASE_SUMMARY_FILEPATH))
+
+    parsed_args = _PARSER.parse_args()
+    if parsed_args.github_username is None:
+        raise Exception(
+            'No GitHub username provided. Please re-run the '
+            'script specifying a username using --username=<Your username>')
+    github_username = parsed_args.github_username
+
+    personal_access_token = getpass.getpass(
+        prompt=(
+            'Please provide personal access token for your github ID. '
+            'You can create one at https://github.com/settings/tokens: '))
+
+    if personal_access_token is None:
+        raise Exception(
+            'No personal access token provided, please set up a personal '
+            'access token at https://github.com/settings/tokens and re-run '
+            'the script')
+    g = github.Github(personal_access_token)
+    repo_fork = g.get_repo('%s/oppia' % github_username)
+
+    current_release_version = branch_name[len(
+        common.RELEASE_BRANCH_NAME_PREFIX):]
+    target_branch = 'update-changelog-for-releasev%s' % current_release_version
+
+    remove_updates_and_delete_branch(repo_fork, target_branch)
+
+    message = (
+        'Please update %s to:\n- have a correct changelog for '
+        'updating the CHANGELOG file\n- have a correct list of new '
+        'authors and contributors to update AUTHORS, CONTRIBUTORS '
+        'and developer_names section in about-page.directive.html\n' % (
+            feconf.RELEASE_SUMMARY_FILEPATH))
+    ask_user_to_confirm(message)
+
+    release_summary_lines = []
+    with python_utils.open_file(
+        feconf.RELEASE_SUMMARY_FILEPATH, 'r') as release_summary_file:
+        release_summary_lines = release_summary_file.readlines()
+
+    check_ordering_of_sections(release_summary_lines)
+
+    update_changelog(
+        release_summary_lines, current_release_version)
+    update_authors(release_summary_lines)
+    update_contributors(release_summary_lines)
+    update_developer_names(release_summary_lines)
+
+    message = (
+        'Please check the changes and make updates if required in the '
+        'following files:\n1. %s\n2. %s\n3. %s\n4. %s\n' % (
+            CHANGELOG_FILEPATH, AUTHORS_FILEPATH, CONTRIBUTORS_FILEPATH,
+            ABOUT_PAGE_FILEPATH))
+    ask_user_to_confirm(message)
+
+    create_branch(repo_fork, target_branch, github_username)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
How should I test this line?

Reply:
See the bottom of scripts/build.py for what we do in this case.

Reply:
Updated according to `scripts/build.py`.